packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_firestore.dart

```diff
@@ -22,155 +21,20 @@ import 'utils/exception.dart';
-      : super(appInstance: app) {
-    if (_initialized) return;
-    channel.setMethodCallHandler((MethodCall call) async {
-      switch (call.method) {
-        case 'Firestore#snapshotsInSync':
-          return _handleSnapshotsInSync(call.arguments);
-          break;
-        case 'QuerySnapshot#event':
-          return _handleQuerySnapshotEvent(call.arguments);
-          break;
-        case 'QuerySnapshot#error':
-          return _handleQuerySnapshotError(call.arguments);
-          break;
-        case 'DocumentSnapshot#event':
-          return _handleDocumentSnapshotEvent(call.arguments);
-          break;
-        case 'DocumentSnapshot#error':
-          return _handleDocumentSnapshotError(call.arguments);
-          break;
-        case 'Transaction#attempt':
-          return _handleTransactionAttempt(call.arguments);
-          break;
-        default:
-          throw UnimplementedError("${call.method} has not been implemented");
-      }
-    });
-    _initialized = true;
-  }
-
-  static int _methodChannelHandleId = 0;
+      : super(appInstance: app);
 
   /// The [Settings] for this [MethodChannelFirebaseFirestore] instance.
   Settings _settings = Settings();
 
-  /// Increments and returns the next channel ID handler for Firestore.
-  static int get nextMethodChannelHandleId => _methodChannelHandleId++;
-
-  /// When a [snapshotsInSync] event is fired on the [MethodChannel], trigger
-  /// a new Stream event.
-  void _handleSnapshotsInSync(Map<dynamic, dynamic> arguments) async {
-    if (!snapshotInSyncObservers.containsKey(arguments['handle'])) {
-      return;
-    }
-
-    snapshotInSyncObservers[arguments['handle']] /*!*/ .add(null);
-  }
-
-  /// When a [QuerySnapshot] event is fired on the [MethodChannel],
-  /// add a [MethodChannelQuerySnapshot] to the [StreamController].
-  void _handleQuerySnapshotEvent(Map<dynamic, dynamic> arguments) async {
-    if (!queryObservers.containsKey(arguments['handle'])) {
-      return;
-    }
-
-    try {
-      queryObservers[arguments['handle']] /*!*/
-          .add(MethodChannelQuerySnapshot(this, arguments['snapshot']));
-    } catch (error) {
-      _handleQuerySnapshotError(<dynamic, dynamic>{
-        'handle': arguments['handle'],
-        'error': error,
-      });
-    }
-  }
-
-  /// When a [QuerySnapshot] error event is fired on the [MethodChannel],
-  /// send the [StreamController] the arguments to throw a [FirebaseException].
-  void _handleQuerySnapshotError(Map<dynamic, dynamic> arguments) {
-    _handleError(queryObservers[arguments['handle']], arguments);
-  }
-
-  /// When a [DocumentSnapshot] event is fired on the [MethodChannel],
-  /// add a [DocumentSnapshotPlatform] to the [StreamController].
-  void _handleDocumentSnapshotEvent(Map<dynamic, dynamic> arguments) async {
-    if (!documentObservers.containsKey(arguments['handle'])) {
+  /// Attach a [FirebaseException] to a given [StreamController].
```

Comment:
I'm not sure what this comment means. I don't see how it pushes a `FirebaseException`